packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -80,7 +87,21 @@ const PdfViewer: FC<Props> = ({
     )
   );
 
-  const canvasInfo = useMemo(() => getCanvasInfo(loadedPage, scale), [loadedPage, scale]);
+  useEffect(
+    () => setCanvasInfo(getCanvasInfo(loadedPage, scale, rootRef)),
+    [loadedPage, scale, rootRef]
+  );
+
+  useResizeObserver(
+    rootRef.current,
+    debounce(() => {
+      const currentRootWidth = rootRef?.current?.getBoundingClientRect().width;
+      if (!!currentRootWidth && currentRootWidth !== previousRootWidth) {
+        setCanvasInfo(getCanvasInfo(loadedPage, scale, rootRef));
+        setPreviousRootWidth(currentRootWidth);
+      }
+    }, RESIZE_DEBOUNCE)
+  );
```

Comment:
I noticed that the author of `useResizeObserver` also has a hook called `useSize` (which uses the former). I think that might be better here. Something like:

```js
const [width] = useSize(rootRef);
useEffect(() => {
  // width has changed; update canvas info
  setCanvasInfo(getCanvasInfo(loadedPage, scale, width)); // NOTE: just pass width, since we know it already
}, [width]);
```

Reply:
Oh, this also takes care of the `useEffect` above this code (since it's the same!).

Reply:
Oh, nice. Didn't notice that, that could be very convenient. I'll check out how well it handles the page resizes (from a debouncing perspective)...

Reply:
If you still need to debounce, you can simply move that to around `setCanvasInfo`.

That author also has a `useDebounce` hook, which may be useful.

Reply:
I think the debounce may not be absolutely necessary-- resizing the window doesn't cause the PDF to jump around too much. We can wait to see if it bothers anyone (unlikely).